src/ui/actions/timeline.ts

```diff
@@ -687,22 +692,38 @@ export function setFocusRegionBeginTime(
     await dispatch(updateDisplayedFocusRegion({ begin, end }));
 
     if (sync) {
-      dispatch(syncFocusedRegion());
+      await dispatch(syncFocusedRegion());
       dispatch(updateFocusRegionParam());
     }
   };
 }
 
 export function syncFocusedRegion(): UIThunkAction {
```

Comment:
I _think_ this method is the single place that we need to add this two-way sync logic.